nall/gdb/server.cpp

```diff
@@ -0,0 +1,535 @@
+        resumeProgram();
+        return "OK";
+
+      case 'D': // client wants to detach (Note: VScode doesn't seem to use this, uses vKill instead)
+        requestDisconnect = true;
+        return "OK";
+      break;
+
+      case 'g': // dump all general registers
+        if(hooks.regReadGeneral) {
+          return hooks.regReadGeneral();
+        } else {
+          return "0000000000000000000000000000000000000000";
+        }
+      break;
+
+      case 'G': // set all general registers
+        if(hooks.regWriteGeneral) {
+          hooks.regWriteGeneral(cmd.slice(1));
+          return "OK";
+        }
+      break;
+
+      case 'H': // set which thread a 'c' command that may follow belongs to (can be ignored in stop-mode)
+        if(cmdName == "Hc0")currentThreadC = 0;
+        if(cmdName == "Hc-1")currentThreadC = -1;
+        return "OK";
+
+      case 'k':  // old version of vKill
+        if(handshakeDone) { // sometimes this gets send during handshake (to reset the program?) -> ignore
+          requestDisconnect = true;
+        }
+        return "OK";
+      break;
+
+      case 'm': // read memory (e.g.: "m80005A00,4")
+        {
+          if(!hooks.read) {
+            return "";
+          }
+
+          auto sepIdxMaybe = cmdName.find(",");
+          u32 sepIdx = sepIdxMaybe ? sepIdxMaybe.get() : 1;
+
+          u64 address = cmdName.slice(1, sepIdx-1).hex();
+          u64 count = cmdName.slice(sepIdx+1, cmdName.size()-sepIdx).hex();
+          return hooks.read(address, count);
+        }
+      break;
+
+      case 'M': // write memory (e.g.: "M801ef90a,4:01000000")
+        {
+          if(!hooks.write) {
+            return "";
+          }
+
+          auto sepIdxMaybe = cmdName.find(",");
+          u32 sepIdx = sepIdxMaybe ? sepIdxMaybe.get() : 1;
+
+          u64 address = cmdName.slice(1, sepIdx-1).hex();
+          u64 unitSize = cmdName.slice(sepIdx+1, 1).hex();
+          u64 value = cmdParts.size() > 1 ? cmdParts[1].hex() : 0;
+
+          hooks.write(address, unitSize, value);
+          return "";
+        }
+
+      break;
+
+      case 'p': // read specific register (e.g.: "p15")
+        if(hooks.regRead) {
+          u32 regIdx = cmdName.slice(1).integer();
+          return hooks.regRead(regIdx);
+        } else {
+          return "00000000";
+        }
+      break;
+
+      case 'P': // write specific register (e.g.: "P15=FFFFFFFF80001234")
+        if(hooks.regWrite) {
+          auto sepIdxMaybe = cmdName.find("=");
+          u32 sepIdx = sepIdxMaybe ? sepIdxMaybe.get() : 1;
+          
+          u32 regIdx = static_cast<u32>(cmdName.slice(1, sepIdx-1).hex());
+          u64 regValue = cmdName.slice(sepIdx+1).hex();
+
+          return hooks.regWrite(regIdx, regValue) ? "OK" : "E00";
+        }
+      break;
+
+      case 'q':
+        // This tells the client what we can and can't do
+        if(cmdName == "qSupported"){ return {
+          "PacketSize=", MAX_PACKET_SIZE, 
```

Comment:
This isn't strictly harmful, but at least LLDB parses this as a hexadecimal value, not a decimal value; might want to verify if that is the case on GDB too.

Reply:
Looking through gdb's sources, it seem to be hex too.. gonna fix that then